### PR TITLE
[MRG, IO] Persyst reader into Raw object

### PR DESCRIPTION
**Pull request by adam2392**
#### Reference issue
Closes: #8173 

#### What does this implement/fix?
Adds a Persyst reader for .lay + .dat file combinations and unit tests.

Note:
- birthdate and measurement date is not a strict adherence sometimes when output by Persyst. I am not sure why... but things like `16-Nov-199` obviously probably stands for nov 16th, 1992 based on the data, but not sure why it gets output funky. I'm assuming these things can change with Persyst versions without our knowing.

With that being said though, the data format and such seems like it shouldn't change all of a sudden. 

#### Additional information
Updates the testing repo to: https://github.com/mne-tools/mne-testing-data/pull/73

### CODE REVIEW

**Comment by adam2392** [comment]
This is ready for review/merge on my end.

I am not sure why that final CI test failed. It says it has to do with unpacking the tar file(?)

File: mne/io/persyst/persyst.py
Comment (agramfort) [inline mne/io/persyst/persyst.py:101]: you already made the key lower above

File: mne/io/persyst/persyst.py
Comment (agramfort) [inline mne/io/persyst/persyst.py:230]: set_annotation in a loop seem weird

Comment (agramfort) [reply mne/io/persyst/persyst.py:230]: are you sure you don't want to do this at the end of the for loop? note that set_annotations flushed previous annotations

File: mne/io/persyst/tests/test_persyst.py
Comment (agramfort) [inline mne/io/persyst/tests/test_persyst.py:21]: you should be calling the _test_raw_reader function on you reader to make sure it's ok.

File: mne/io/persyst/persyst.py
Comment (adam2392) [reply mne/io/persyst/persyst.py:230]: Ah yes that makes sense. You're right. I think I might've indented at some point w/o realizing it.

File: mne/io/persyst/persyst.py
Comment (larsoner) [inline mne/io/persyst/persyst.py:217]:
```suggestion
        num_comments = len(comments_dict)
```

File: mne/io/persyst/persyst.py
Comment (larsoner) [inline mne/io/persyst/persyst.py:277]: `mult` is unused, so this code is going to be broken when projectors are used. You probably want to let `_mult_cal_one` take care of this stuff.

Also if you set `info['chs'][k]['cal']` appropriately above, you don't need to do the `1e-6` mult (here done as a div by `10 ** 6`) yourself. Setting it to 1e6 or 1e-6 (can't remember which) should have an equivalent effect.

File: mne/io/persyst/tests/test_persyst.py
Comment (larsoner) [inline mne/io/persyst/tests/test_persyst.py:84]: This is the old way to do it. Better to make use of the pytest fixture:
```suggestion
    out_dir = mne.utils._TempDir()
    out_dir = str(tmpdir)
```

File: mne/io/persyst/tests/test_persyst.py
Comment (larsoner) [inline mne/io/persyst/tests/test_persyst.py:79]:
```suggestion
def test_persyst_wrong_file(tmpdir):
```

File: mne/io/persyst/persyst.py
Comment (adam2392) [reply mne/io/persyst/persyst.py:79]: Yes these are the order, but they're always in the increasing order that they are read in. No reason to suspect that the order would change unless the user explicitly modified the file, which shouldn't be done. WDYT? worth adding?

File: mne/io/persyst/persyst.py
Comment (adam2392) [reply mne/io/persyst/persyst.py:80]: Yeah sampletimes seems redundant, and I included it in here because it was a field, but you can obtain the same info via the `testtime` and `testdate`.

File: mne/io/persyst/persyst.py
Comment (adam2392) [reply mne/io/persyst/persyst.py:262]: Okay I see, I'll adjust to multiply the info cal accordingly.

Re `mult_cal_one` sorry what is this? I've never worked w/ projectors :p

Comment (larsoner) [reply mne/io/persyst/persyst.py]: Basically instead of multiplying by `cals` and sub-indexing yourself you should let `_mult_cal_one` do it. See how it's used in other readers by `git grep _mult_cal_one`.

**Review by agramfort — APPROVED** [review]
looks good !

@adam2392 you should also see then how mne-bids allows you to write such files in bids

**Comment by adam2392** [comment]
This should be good to go assuming the CIs turn green and @larsoner is okay with my changes to address his comments :). Thanks!

**Comment by adam2392** [comment]
Just wanted to confirm that this is "good to go", possibly waiting for some additional CI test from @larsoner in #8210?

**Comment by larsoner** [comment]
Yes let's wait for #8210. I just got it working (I think!). Do you want me to push a merge commit once #8210 is in `master` so we can see if everything is good here?

File: mne/io/persyst/persyst.py
Comment (larsoner) [inline mne/io/persyst/persyst.py:195]: No need to nest in `if verbose`, it magically gets accounted for
```suggestion
            logger.debug(f'Loaded {n_samples} samples '
```

File: mne/io/persyst/persyst.py
Comment (larsoner) [inline mne/io/persyst/persyst.py:290]: I would use a dict lookup to avoid all of these uncovered lines and make it easier to expand later:
```
sex_dict = dict(
    m=...,
    male=...,
    f=...,
    female=...,
)
subject_info['sex'] = sex_dict.get(subject_info['sex'], FIFF.FIFFV_SEX_UNKNOWN)
```

File: mne/io/persyst/persyst.py
Comment (larsoner) [inline mne/io/persyst/persyst.py:305]: same here but I would do:
```
try:
    subject_info['hand'] = hand_dict[subject_info['hand']]
except KeyError:
    subject_info.pop('hand')

**Comment by larsoner** [comment]
Looks like the channel with index 55 (name `'pol $tp9'`) has values all right around `[-0.01200273]`, was this a bad channel during the recording or something?

**Comment by adam2392** [comment]
> Looks like the channel with index 55 (name `'pol $tp9'`) has values all right around `[-0.01200273]`, was this a bad channel during the recording or something?

That is a byproduct of some center's recording systems I work with. So a rule of thumb I've come to realize is that if there is a `$` sign, then it most likely means its a dead channel (not connected, or used as REF, etc.)

**Review by agramfort — APPROVED** [review]
@larsoner merge if you're happy

**Comment by larsoner** [comment]
Thanks @adam2392 !

**Pull request merged**
